packages/terra-functional-testing/src/config/wdio.conf.js

```diff
@@ -165,4 +166,17 @@ exports.config = {
     ui: 'bdd',
     timeout: 60000,
   },
+
+  /* Terra defined viewport sizes. */
+  terraViewports: {
+    tiny: { width: 470, height: 768, name: 'tiny' },
+    small: { width: 622, height: 768, name: 'small' },
+    medium: { width: 838, height: 768, name: 'medium' },
+    large: { width: 1000, height: 768, name: 'large' },
+    huge: { width: 1300, height: 768, name: 'huge' },
+    enormous: { width: 1500, height: 768, name: 'enormous' },
+  },
+
+  /* Use to change the form factor (test viewport) used in the wdio run. */
+  ...FORM_FACTOR && { formFactor: FORM_FACTOR },
```

Comment:
These should be scoped to the service they are a part of, either as options or as nested keys. There are trade offs to both, but I think we should avoid un-scoped flat keys on the configuration object.

Service Options:
```js
  services: [
    [TerraService, {
      viewports: {
        tiny: { width: 470, height: 768, name: 'tiny' },
        small: { width: 622, height: 768, name: 'small' },
        medium: { width: 838, height: 768, name: 'medium' },
        large: { width: 1000, height: 768, name: 'large' },
        huge: { width: 1300, height: 768, name: 'huge' },
        enormous: { width: 1500, height: 768, name: 'enormous' },
      },
      ...FORM_FACTOR && { formFactor: FORM_FACTOR },
    }],
  ],
```

Scoped keys:

```
// wdio.config.js

module.exports = {
  terraServiceOptions: {
      viewports: {
        tiny: { width: 470, height: 768, name: 'tiny' },
        small: { width: 622, height: 768, name: 'small' },
        medium: { width: 838, height: 768, name: 'medium' },
        large: { width: 1000, height: 768, name: 'large' },
        huge: { width: 1300, height: 768, name: 'huge' },
        enormous: { width: 1500, height: 768, name: 'enormous' },
      },
      ...FORM_FACTOR && { formFactor: FORM_FACTOR },
  }
}
```
I'll work on going back through previous conversations and linking them here to show some of the tradeoffs/concerns. The most challenging scenario is how do consumers extend and override service options.

Reply:
Updated: https://github.com/cerner/terra-toolkit/pull/438/commits/3db345e69d9b12382795557acce0285b0bd8abbc